File: src/main/java/uk/co/lukestevens/geoguessr/models/ApiCredentials.java
package uk.co.lukestevens.geoguessr.models;

import javax.persistence.*;

@Entity
@Table(name = "api_credentials", schema = "geoguessr")
public class ApiCredentials {

    private static final String COOKIE_HEADER_TEMPLATE = "devicetoken=%s; _ncfa=%s";

    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    @Column(name = "credentials_id")
    private Long id;

    @Column(name = "username")
    private String user;

    @Column(name = "device_token")
    private String deviceToken;

    @Column(name = "ncfa")
    private String ncfa;


    public String getCookieHeader(){
        return String.format(COOKIE_HEADER_TEMPLATE, deviceToken, ncfa);
    }

    public void setId(Long id) {
        this.id = id;
    }

    public void setUser(String user) {
        this.user = user;
    }

    public void setDeviceToken(String deviceToken) {
        this.deviceToken = deviceToken;
    }

    public void setNcfa(String ncfa) {
        this.ncfa = ncfa;
    }
}

File: src/main/kotlin/uk/co/lukestevens/geoguessr/models/ApiCredentials.kt
package uk.co.lukestevens.geoguessr.models

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.Table

@Entity
@Table(name = "api_credentials", schema = "geoguessr")
class ApiCredentials {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "credentials_id")
    var id: Long? = null

    @Column(name = "username")
    var user: String? = null

    @Column(name = "device_token")
    var deviceToken: String? = null

    @Column(name = "ncfa")
    var ncfa: String? = null

    val cookieHeader: String
        get() = String.format(COOKIE_HEADER_TEMPLATE, deviceToken, ncfa)

    companion object {
        private const val COOKIE_HEADER_TEMPLATE = "devicetoken=%s; _ncfa=%s"
    }
}